src/app/actions.ts

```diff
@@ -19,23 +17,43 @@ export type FlattenedRegion = {
   fullHierarchyName: string;
 };
 
-const fuseOptions = {
+export async function getRegion(regionCode: string) {
+  // We're reassigning the imported flattenedRegionCodes to a new variable
+  // here exclusively so that we can cast it to a Region[] type and get
+  // good type checking in the rest of the function.
+  const regions = flattenedRegionCodes as Region[];
+
+  const region = regions.find((region) => region.code === regionCode);
+
+  if (!region) {
+    throw new Error(`Could not find region with code ${regionCode}`);
+  }
+
+  return region;
+}
```

Comment:
Great! 
I had a random thought earlier about how I'd mentioned using Fuse again for this but of course it's an exact match we're after. So using .find makes perfect sense.

Reply:
Yep! You could probably use Fuse and crank it's matching algo _way_ up, it would probably be faster as well tbh, but this is simple and fast _enough_ for what we're doing. 

I think at some point this will become slow but at that point I think we'll move it into an actual DB with an index on the region code so it's _lightning fast_

Reply:
Yeah, great! DB shenanigans are in the pipeline, we'll get there.